### PR TITLE
Add custom clocking example to docstring of Clock().

### LINKED ISSUE #1158: Setting arbitrary clock phase
Is there a convenient way to generate a clock with an arbitrary clock phase? From [this pr](https://github.com/cocotb/cocotb/pull/1036) it looks like it's possible to set the clock polarity. Is it possible to generalize this and allow arbitrary phase setting? One example where this would be useful is when simulating clocks synchronized via a PLL.

### PR DESCRIPTION
**Pull request by cmarqu**
Closes #1158.

### CODE REVIEW

File: cocotb/clock.py [hunk 1]
```diff
@@ -65,12 +65,43 @@ class Clock(BaseClock):
...
+    If you need more features like a phase shift and an asymmetric duty cycle,
+    it is simple to create your own clock generator (that you then :func:`fork`):
+
+    .. code-block:: python
+
+        @cocotb.coroutine
+        def custom_clock():
+            high_time = Timer(high_delay)
+            low_time = Timer(low_delay)
+            yield Timer(initial_delay)
```
Comment (eric-wieser) [inline cocotb/clock.py]: Perhaps add `# pre-construct triggers for performance`

File: cocotb/clock.py [hunk 2]
```diff
@@ -65,12 +65,43 @@ class Clock(BaseClock):
...
+    .. code-block:: python
+
+        @cocotb.coroutine
+        def custom_clock():
+            yield Timer(initial_delay)
+            while True:
+                dut.clk <= 1
+                yield Timer(high_delay)
+                dut.clk <= 0
+                yield Timer(low_delay)
```
Comment (eric-wieser) [inline cocotb/clock.py]: Not sure this is worth mentioning - without showing the other coroutine changing these parameters its quite hard to understand the difference.

File: cocotb/clock.py [hunk 3]
```diff
@@ -83,12 +114,12 @@ def __init__(self, signal, period, units=None):
 
     @cocotb.coroutine
     def start(self, cycles=None, start_high=True):
-        """Clocking coroutine.  Start driving your clock by forking a 
+        """Clocking coroutine.  Start driving your clock by :func:`fork`\ ing a
```
Comment (eric-wieser) [inline cocotb/clock.py]:
```suggestion
        r"""Clocking coroutine.  Start driving your clock by :func:`fork`\ ing a
```

File: cocotb/clock.py (hunk 1 above)
Comment (cmarqu) [reply cocotb/clock.py]: Done.

File: cocotb/clock.py (hunk 2 above)
Comment (cmarqu) [reply cocotb/clock.py]: I have extended the comment above the code a bit to make it clearer.

File: cocotb/clock.py (hunk 3 above)
Comment (cmarqu) [reply cocotb/clock.py]: Done. Thanks for the review!

File: cocotb/clock.py [hunk 4]
```diff
@@ -65,12 +65,46 @@ class Clock(BaseClock):
...
+    .. code-block:: python
+
+        @cocotb.coroutine
+        def custom_clock():
+            yield Timer(initial_delay)
+            while True:
+                dut.clk <= 1
+                yield Timer(high_delay)
+                dut.clk <= 0
+                yield Timer(low_delay)
```
Comment (eric-wieser) [inline cocotb/clock.py]: I still think this is probably hard to follow. Perhaps give the larger example:
```
        @cocotb.coroutine
        def custom_clock():
            while True:
                dut.clk <= 1
                yield Timer(high_delay)
                dut.clk <= 0
                yield Timer(low_delay)

        high_delay = low_delay = 100
        cocotb.fork(custom_clock())
        yield Timer(1000)
        high_delay = low_delay = 10  # change the clock speed
        yield Timer(1000)
```
Note that I dropped initial_delay, because changing that at runtime doesn't make sense.

File: cocotb/clock.py [hunk 5]
```diff
@@ -65,12 +65,46 @@ class Clock(BaseClock):
...
+    If you need more features like a phase shift and an asymmetric duty cycle,
+    it is simple to create your own clock generator (that you then :func:`fork`):
+
+    .. code-block:: python
+
+        @cocotb.coroutine
+        def custom_clock():
+            # pre-construct triggers for performance
+            high_time = Timer(high_delay)
+            low_time = Timer(low_delay)
```
Comment (eric-wieser) [inline cocotb/clock.py]: Didn't you make an issue about always using units in `Timer` examples?

Comment (cmarqu) [reply cocotb/clock.py]: Right, I just noticed that too.

File: cocotb/clock.py (hunk 4 above)
Comment (cmarqu) [reply cocotb/clock.py]: Alright, next version. :)

**Comment by eric-wieser** [comment]
Close/reopen to build docs

**Comment by eric-wieser** [comment]
@cmarqu: [Rendered output](https://external-builds.readthedocs.io/html/cocotb/1172/library_reference.html#cocotb.clock.Clock)!

File: cocotb/clock.py [hunk 6]
```diff
@@ -50,7 +50,7 @@ def log(self):
 class Clock(BaseClock):
     """Simple 50:50 duty cycle clock driver.
 
-    Instances of this class should call its :meth:`start` method and fork the
+    Instances of this class should call its :meth:`start` method and :func:`fork` the
```
Comment (eric-wieser) [inline cocotb/clock.py]: This doesn't actually produce a link, but that can be fixed in a follow up. Perhaps we have no documentation for `fork`?

**Comment by cmarqu** [comment]
> @cmarqu: [Rendered output](https://external-builds.readthedocs.io/html/cocotb/1172/library_reference.html#cocotb.clock.Clock)!

Yes, cool, I have seen it :) Just yesterday I set up the build process for this branch in my fork's RTD, but I guess I can retire that now.

File: cocotb/clock.py (hunk 6 above)
Comment (cmarqu) [reply cocotb/clock.py]: Yes, ``fork`` is missing a docstring, I opened #1176 for that. Could you suggest a docstring there? I'm not sure I'd get the wording nice.

File: cocotb/clock.py [hunk 7]
```diff
@@ -65,12 +65,51 @@ class Clock(BaseClock):
...
+            yield Timer(initial_delay, units="ns")
+            while True:
+                dut.clk <= 1
+                yield high_time
+                dut.clk <= 0
+                yield low_time
+
+    If you also want to change the timing during simulation,
+    use this slightly more inefficient example instead where 
+    the Timers inside the while loop are created with
```
Comment (eric-wieser) [inline cocotb/clock.py]:
```suggestion
    the :class:`Timer`\ s inside the while loop are created with 
```

Comment (cmarqu) [reply cocotb/clock.py]: Done.

File: cocotb/clock.py [hunk 8]
```diff
@@ -65,12 +65,51 @@ class Clock(BaseClock):
...
+            yield Timer(initial_delay, units="ns")
+            while True:
+                dut.clk <= 1
+                yield high_time
+                dut.clk <= 0
+                yield low_time
+
+    If you also want to change the timing during simulation,
+    use this slightly more inefficient example instead where 
+    the :class:`Timer`\ s inside the while loop are created with
```
Comment (imphil) [inline cocotb/clock.py]: The flake8 check in CI fails here: 
`./cocotb/clock.py:93:23: W605 invalid escape sequence '\ '` (https://travis-ci.org/cocotb/cocotb/jobs/612518805?utm_medium=notification&utm_source=github_status)

Comment (eric-wieser) [reply cocotb/clock.py]: Needs either a double backslash, or an `r` on the docstring

Comment (eric-wieser) [reply cocotb/clock.py]: Fixed it myself
